govcd/edgegateway_test.go

```diff
@@ -5,192 +5,117 @@
+func (vcd *TestVCD) Test_Refresh(check *C) {
+	if vcd.config.VCD.EdgeGateway == "" {
+		check.Skip(fmt.Sprintf("skipping %s because no edgegatway given", check.TestName()))
+	}
+	edge, err := vcd.vdc.FindEdgeGateway(vcd.config.VCD.EdgeGateway)
+	check.Assert(err, IsNil)
+	check.Assert(edge.EdgeGateway.Name, Equals, vcd.config.VCD.EdgeGateway)
 	err = edge.Refresh()
-	_ = testServer.WaitRequest()
-	testServer.Flush()
-
-	c.Assert(err, IsNil)
-	c.Assert(edge.EdgeGateway.Name, Equals, "M916272752-5793")
-
+	check.Assert(err, IsNil)
+	check.Assert(edge.EdgeGateway.Name, Equals, vcd.config.VCD.EdgeGateway)
 }
 
-func (vcd *TestVCD) Test_NATMapping(c *C) {
-	testServer.ResponseMap(2, testutil.ResponseMap{
-		"/api/vdc/00000000-0000-0000-0000-000000000000/edgeGateways":  testutil.Response{200, nil, edgegatewayqueryresultsExample},
-		"/api/admin/edgeGateway/00000000-0000-0000-0000-000000000000": testutil.Response{200, nil, edgegatewayExample},
-	})
-
-	edge, err := vcd.vdc.FindEdgeGateway("M916272752-5793")
-	_ = testServer.WaitRequests(2)
-	testServer.Flush()
-
-	c.Assert(err, IsNil)
-	c.Assert(edge.EdgeGateway.Name, Equals, "M916272752-5793")
-
-	testServer.ResponseMap(2, testutil.ResponseMap{
-		"/api/admin/edgeGateway/00000000-0000-0000-0000-000000000000":                          testutil.Response{200, nil, edgegatewayExample},
-		"/api/admin/edgeGateway/00000000-0000-0000-0000-000000000000/action/configureServices": testutil.Response{200, nil, taskExample},
-	})
-
-	_, err = edge.AddNATMapping("DNAT", "10.0.0.1", "20.0.0.2", "77")
-	_ = testServer.WaitRequest()
-
-	c.Assert(err, IsNil)
-
-	testServer.ResponseMap(2, testutil.ResponseMap{
-		"/api/admin/edgeGateway/00000000-0000-0000-0000-000000000000":                          testutil.Response{200, nil, edgegatewayExample},
-		"/api/admin/edgeGateway/00000000-0000-0000-0000-000000000000/action/configureServices": testutil.Response{200, nil, taskExample},
-	})
-
-	_, err = edge.RemoveNATMapping("DNAT", "10.0.0.1", "20.0.0.2", "77")
-	_ = testServer.WaitRequest()
-
-	c.Assert(err, IsNil)
-
+func (vcd *TestVCD) Test_NATMapping(check *C) {
+	if vcd.config.VCD.Externalip == "" || vcd.config.VCD.Internalip == "" {
+		check.Skip(fmt.Sprintf("skipping %s because no valid ip given", check.TestName()))
+	}
+	if vcd.config.VCD.EdgeGateway == "" {
+		check.Skip(fmt.Sprintf("skipping %s because no edgegatway given", check.TestName()))
+	}
+	edge, err := vcd.vdc.FindEdgeGateway(vcd.config.VCD.EdgeGateway)
+	check.Assert(err, IsNil)
+	check.Assert(edge.EdgeGateway.Name, Equals, vcd.config.VCD.EdgeGateway)
+
+	task, err := edge.AddNATMapping("DNAT", vcd.config.VCD.Externalip, vcd.config.VCD.Internalip, "77")
+	check.Assert(err, IsNil)
+	err = task.WaitTaskCompletion()
+	check.Assert(err, IsNil)
+	task, err = edge.RemoveNATMapping("DNAT", vcd.config.VCD.Externalip, vcd.config.VCD.Internalip, "77")
+	check.Assert(err, IsNil)
+	err = task.WaitTaskCompletion()
+	check.Assert(err, IsNil)
```

Comment:
There should be a check of the final state of the mapping. (Future test enhancement to be marked in a TODO comment)